trino/trino.go

```diff
@@ -963,9 +979,12 @@ func (st *driverStmt) exec(ctx context.Context, args []driver.NamedValue) (*stmt
 	}
 
 	var cancel context.CancelFunc = func() {}
-	if _, ok := ctx.Deadline(); !ok {
+	if st.conn.timeout != nil {
+		ctx, cancel = context.WithTimeout(ctx, *st.conn.timeout)
+	} else if _, ok := ctx.Deadline(); !ok {
```

Comment:
Both timeouts should always apply

Reply:
Not sure I understand, if both timeouts apply wouldn't this be capped at 60s? Should I increase the default in that case?

Reply:
If a user sets timeouts both in the connection string, and in the context, we should never silently ignore one of them. We should choose the smaller one. The default only applies when no timeout is set.

I do think we should increase the default significantly, maybe even to 10 hours. Small timeouts are fine for things like network requests, which can get stuck, and can be retried. We're in no position to assume there's anything wrong with a query that runs for a long time.

Reply:
Ah that's already how your changes work - I think adding a timeout to a context with an existing deadline will cause one to trigger first anyway.